docs/event.md

```diff
@@ -0,0 +1,93 @@
+- I've tried to keep terminology human understable - Eg, the removal of 'category', 'key', 'value', 'data' - as these should all be inherent in the data structure etc. Anything bespoke goes in 'meta'.
+- It is intentionally abstracted away from any particular database format and meant to represent a logical, easy to understand data structure.
+- It assumes the consumers of the event stream can manipulate the data in to their desired storage format.
+- It doesn't assume users of the data will need to join the data across tables to find the information they want (a downside of the parent-child relationship). Though consumers may wish to split this data in to their own tables for query optimisation etc.
+
+## Spec 
+
+`event.source` and `event.api-key` are MANDATORY, everything else is OPTIONAL
+
+This allows a low barrier to entry for anyone wanting to submit events to Spoor.
+
+## Enrichments 
+
+Each payload will be scanned for the presence of these properties.
+
+If present the properties will be decorated and/or tokenised. 
+
+To date we've discussed :-
+
+- If `user.session.token` is included it will be validated against the session api and expanded in to a membership uuid.
+- If `content.uuid` is included it will be expanded in to a set of content api meta-data.
+- ...
+
+## Format
+
+```
+{
+    "event": {
+        "source": "email.simple-email-service",  				// Name of the sender's system [1]
+        "type": "opened",  										// The type of event that happened, Eg. 'pageview', 'dwell', 'play' etc.
+		"api-key":	"0f7464b4-3f4d-11e4-984b-00144feabdc0"		// Sender-specific key [6]
+    },
+    "user": {													// Who generate the event
+        "user-agent": "Mozilla ...",
+        "ip": "123.123.123.123",
+        "anonymous-id": "...",									// An anonymous token, supplied by the client [2] 
+        "session": {
+            "token": "asdf324dfag1ds%asdf1A-1sadsadf",  		// FT Session token - expanded in to a uuid and valid flag [3]
+        },
+		"device": {
+			"layout": "S",										// require('o-grid').getCurrentLayout 
+        	"orientation": "portrait"							// require('o-viewport').getOrientation
+		},
+		"geo": {												// [8]
+            "country": "MA",
+            "area_code": "0",
+            "longitude": "-5.000",
+            "latitude": "32.000",
+            "country_name": "Morocco",
+            "continent": "AF"
+		},
+		"ab": {
+			"test-name": "variant"								// [9]
+		},
+		"flags": [ ],											// Flags/Toggles the user has set [10]
+		"isStaff": true											// [9]
+    },
+    "content": {
+        "uuid": "", 				// Content uuid [4] 
+        "url": "...", 	 			// [5] 
+        "referrer": "...", 			// See also [5]
+		"type":	"content"			// Some sort of controlled list  
+    },
+	"meta": {						// [7]
+		"anything": "goes here",
+		"could: [ "be", "an", "array" ],
+		"or" : { "might": { "be": ["an", "object"] }}
+	},
+    "time": {
+         "offset": 234				// For offline stuff, I think
```

Comment:
Not just that, but any failed request, which will be queued by o-tracking and resent with an updated offset value.